client/internal/engine.go

```diff
@@ -205,8 +205,7 @@ func (e *Engine) Start() error {
 			go e.mobileDep.DnsReadyListener.OnReady()
 		}
 	} else if e.dnsServer == nil {
-		// todo fix custom address
-		e.dnsServer, err = dns.NewDefaultServer(e.ctx, e.wgInterface, e.config.CustomDNSAddress)
+		e.dnsServer, err = dns.NewDefaultServer(e.ctx, e.wgInterface, e.config.CustomDNSAddress, e.mobileDep.InterfaceName, wgAddr)
```

Comment:
I do not think so we should pass the e.mobileDep.InterfaceName and wgAddr. The e.wgInterface knows these informations. But regarding the interface name in the engine config and mobileDep will be different name what is incorrect. So remove the interfaceName from the MobileDependency and overwrite the config.WgIface (engine config) in the connect.go:Run function

Reply:
I had some trouble with getting data from the wgInterface that it was empty. I will try to overwrite it